akka-actor/src/main/scala/akka/serialization/Serializer.scala

```diff
@@ -76,6 +76,15 @@ trait Serializer {
   final def fromBinary(bytes: Array[Byte], clazz: Class[_]): AnyRef = fromBinary(bytes, Option(clazz))
 }
 
+object Serializer {
+
+  def manifestFor(s: Serializer, message: AnyRef): Option[String] = s match {
```

Comment:
I've decided to remove duplicates and thought that it can be in the Serializer object, but probably it is better to create this method in the Serializer trait. WDYT?

Reply:
We can't change `trait Serializer` because of backwards compatibility reason.

Reply:
I don't know if it will be a great solution, but we can have a default implementation in the `trait Serializer` wich can be overriden, as I understand it won't break compatibility

Reply:
> I don't know if it will be a great solution, but we can have a default implementation in the trait Serializer wich can be overriden, as I understand it won't break compatibility

This probably won't work for the custom `Serializers`